### PR TITLE
blockchain/v2: send status request when new peer joins

### LINKED ISSUE #5766: blockchain/v2: broadcast status request after connectivity to the network is restored
e2e tests failure: https://github.com/tendermint/tendermint/commit/607565f34eb47e3796bfa21ad78930158b88dbf5/checks

Currently, blockchain/v2 broadcasts status requests every 10s. This means if the node is fast syncing and had N=0 peers (they all crashed or something), when the node finally has some peers (N>0) **the wait time can be up to 10s** before we learn about the peers (what height they are on) and able to request blocks from them.

I'd suggest we send `StatusRequest` (along with `StatusResponse`) in `AddPeer`. That way, when a new peer joins, the node "immediately" learns what height the new peer is on and can request blocks from it.

### PR DESCRIPTION
**Pull request by melekes**
Closes #5766

### CODE REVIEW

File: blockchain/v2/reactor.go [hunk 1]
```diff
@@ -382,7 +385,10 @@ func (r *BlockchainReactor) demux(events <-chan Event) {
 				r.processor.send(event)
 				r.scheduler.stop()
 			case scSchedulerFail:
-				r.logger.Error("Scheduler failure", "err", event.reason.Error())
+				if scSchedulerFailErr != event.reason {
+					r.logger.Error("Scheduler failure", "err", event.reason.Error())
+					scSchedulerFailErr = event.reason
+				}
```
Comment (alexanderbez) [inline blockchain/v2/reactor.go:391]: This is so we just don't keep logging the same thing right?

**Review by alexanderbez — APPROVED** [review]

File: blockchain/v2/reactor.go (hunk 1 above)
Comment (melekes) [reply blockchain/v2/reactor.go:391]: yep

**Review by marbar3778 — COMMENTED** [review]
🔥

**Review by marbar3778 — APPROVED** [review]

File: blockchain/v2/io.go [hunk 2]
```diff
@@ -132,6 +133,24 @@ func (sio *switchIO) trySwitchToConsensus(state state.State, skipWAL bool) bool
 	return ok
 }
 
+func (sio *switchIO) sendStatusRequest(peerID p2p.ID) error {
```
Comment (erikgrinaker) [inline blockchain/v2/io.go:136]: Maybe just pass the peer directly, so we don't have to look it up in the switch.

**Review by erikgrinaker — APPROVED** [review]

**Pull request merged**